draft-ietf-quic-multipath.md

```diff
@@ -620,6 +620,23 @@ to abandon a path for any reason, for example, removing a hole from
 the sequence of Path IDs in use. This is not an error. The endpoint that
 receive such a PATH_ABANDON frame must treat it as specified in {{path-close}}.
 
+### Closing Duplicate Paths
+
+As noted in {{basic-design-points}}, it is possible to create paths that
+refer to the same 4-tuple. There will be cases where this is intentional,
+for example if the paths use different Differentiated Service markings.
+There may also be cases where this is not intentional, maybe as a result
+of path migration.
+
+In the non-intentional case, the endpoints
+may want to abandon one of the paths and keep the other, but
+uncoordinated Abandon from both ends of the connection may result in deleting
+two paths instead of just one. This sort of error will be avoided if only
+the client initiates the closing of duplicate paths. Applications may override
```

Comment:
There are other ways to avoid problems.  For instance, you might institute a rule that the lowest-numbered path should be closed in the case of duplicates.  This "if only" isn't "only if", but maybe avoid using that sort of phrasing to avoid creating confusion about what you mean.

It's also unclear whether you are recommending that clients be responsible for path closure or whether this is a "for example".  That's a specific manifestation of the more general problem that we don't really know what processes drive the creation and teardown of paths.  If you lack that broader framework, the best you can do is this sort of non-specific advice.

Reply:
Yes. I think that the creation of paths ought to be controlled by the application. Stacks could read what alternate IP addresses are available and just try them all, but the failure modes are easy to imagine. Plus, applications will want to specify the scheduling of packet on paths, for example setting affinity between stream and paths.

The problem with rules like "lowest number first" is that the two endpoints need to agree on the rule. In our current draft, paths are always created by the client. That does not mean that servers should never abandon paths, but they should do that mostly if the path appears broken. In the duplicate example, the two paths are both valid, so there is no obvious criteria. Letting the client chose seems the simplest solution.

Reply:
But yes, the sentence could be written better.

Reply:
To add to this discussion. I also think a rule like lowest number path or actually oldest path would make more sense. If the client don't want to have the new path, it would not have opened it. however, it make open the new path and then plan to close the old one for some reason. On the other hand, if the old one is in active use and the new one not yet, that it's maybe not optional to close the old. So in summary not sure if we can really give good advise here or just simply with down these consideration (and implicitly recommend to not close any of the paths)?

Reply:
I think I fixed that when moving the text to the implementation section.